tests/generic_config_updater/gu_utils.py

```diff
@@ -238,3 +238,31 @@ def rollback_or_reload(duthost, cp=DEFAULT_CHECKPOINT_NAME):
     if output['rc'] or "Config rolled back successfull" not in output['stdout']:
         config_reload(duthost)
         pytest.fail("config rollback failed. Restored by config_reload")
+
+def check_show_ip_intf(duthost, intf_name, expected_content_list, unexpected_content_list, is_ipv4=True):
+    """Check lo interface status by show command
+
+    Sample output:
+    admin@vlab-01:~$ show ip interfaces  | grep Vlan1000
+    Vlan1000                   192.168.0.1/21       up/up         N/A             N/A
+    admin@vlab-01:~$ show ipv6 interfaces | grep Vlan1000
+    Vlan1000                          fc02:1000::1/64                             up/up         N/A             N/A
+                                      fe80::5054:ff:feda:c6af%Vlan1000/64                       N/A             N/A
+    """
+    address_family = "ip" if is_ipv4 else "ipv6"
+    output = duthost.shell("show {} interfaces | grep {} || true".format(address_family, intf_name))
```

Comment:
>true [](http://example.com/codeflow?start=60&length=4)

Any chance the command will fail if you don't add `true` ?

Reply:
If grep no output, it will return a failure with error code 1. If there is a scenario we grep a non-existed variable, we can check if output is empty instead of fail.

Reply:
Catching return code 1 is better to ignore any return code. The whole command could fail with other reasons such as some container stopped (so failed before first pipe), and they will lead to different return code.

Reply:
Please see here https://github.com/Azure/sonic-mgmt/pull/4835#discussion_r774870558
Seems we can only capture the error of the last pipe command.

> The whole command could fail with other reasons such as some container stopped (so failed before first pipe), and they will lead to different return code.

Even though the command fail before first pipe, the return code is only related to the last pipe.